pages/client-libraries/rust.mdx

```diff
@@ -0,0 +1,714 @@
+                value => println!("Value: {}", value),
+            }
+        }
+    }
+}
+
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+Method executed takes the following arguments:
+
+- `query` - The query that will be executed.
+- `params` - The parameters that will be passed to the query.
+
+
+If you do not need to fetch the results, you can use the `execute_without_results` method, then the code becomes simpler:
+ 
+```rust
+let _create_node = "CREATE (n:Technology {name: 'Memgraph'}) RETURN n";
+connection.execute_without_results(_create_node).unwrap();
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+`execute_without_results` method takes just the query as an argument.
+
+#### Run a read query   
+
+The following query will read data from the database: 
+
+```rust
+let _read_node = "MATCH (n:Technology {name: 'Memgraph'}) RETURN n";
+let _columns = connection.execute(_read_node, None);
+while let Ok(result) =  connection.fetchall() {
+    for record in result {
+        for value in record.values {
+            match value {
+                Value::Node(node) => println!("Node: {}", node),
+                value => println!("Value: {}", value),
+            }
+        }
+    }
+}
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+In this example, the match statement is used to distinguish between different types of values that can be returned from the database. In this case the query will return a node, so the rest of logic could be based on that information. 
+
+
+#### Running a queries with property map 
+
+If you want to pass a property map to the query, you can do it like this:
+
+```rust
+let _create_node = "CREATE (n:Technology {name: $name, description: $description}) RETURN n";
+let mut params = HashMap::new();
+params.insert("name".to_string(), QueryParam::String("Memgraph".to_string()));
+params.insert("description".to_string(), QueryParam::String("Fastest graph DB in the world!".to_string()));
+let _columns = connection.execute(_create_node, Some(&params));
+while let Ok(result) =  connection.fetchall() {
+    for record in result {
+        for value in record.values {
+            match value {
+                Value::Node(node) => println!("Node: {}", node),
+                value => println!("Value: {}", value),
+            }
+        }
+    }
+}
+if let Err(e) = connection.commit() {
+    println!("Error: {}", e);
+}
+```
+
+Using this approach, the queries will not contain hard-coded values, they can be more dynamic.
+
+
+### Process the results
+
+In order to read results and serve them back to you Rust application you need to handle them. Rust is statically typed language and proper types need to be handled. 
+Depending on the type of the request you made, you can get different results back. Lets go over a few basic examples on how to handle different types and access properties of the returned results.
```

Comment:
```suggestion
In order to serve the read results back to the Rust application, their types need to be handled properly because Rust is a statically typed language. 
Depending on the type of request made, you can receive different results. Let's go over a few basic examples of how to handle different types and access properties of the returned results.
```